test/services/smart-action-form-layout-service.test.js

```diff
@@ -0,0 +1,263 @@
+const { init, inject } = require('@forestadmin/context');
+const SmartActionFormLayoutService = require('../../src/services/smart-action-form-layout-service');
+
+const setup = () => {
+  init((context) => context
+    .addUsingClass('smartActionFormLayoutService', () => SmartActionFormLayoutService));
+
+  return inject();
+};
+
+describe('services > smart-action-form-layout', () => {
+  describe('validateLayoutElement', () => {
+    describe('errors', () => {
+      it('should throw with message when layout element has an unexpected type', async () => {
+        expect(() => SmartActionFormLayoutService.validateLayoutElement({ component: 'horizontal-separator' })).toThrow('horizontal-separator is not a valid component. Valid components are Row or Page or Separator or HtmlBlock');
+      });
+
+      it('should throw with message when Page does not contain elements', async () => {
+        expect(() => SmartActionFormLayoutService.validateLayoutElement({ component: 'Page' })).toThrow('Page components must contain an array of fields or layout elements in property \'elements\'');
+      });
+
+      it('should throw with message when Row does not contain fields', async () => {
+        expect(() => SmartActionFormLayoutService.validateLayoutElement({ component: 'Row' })).toThrow('Row components must contain an array of fields in property \'fields\'');
+      });
+      it('should throw with message when Row contains layout elements', async () => {
+        expect(() => SmartActionFormLayoutService.validateLayoutElement({ component: 'Row', fields: [{ type: 'Layout', component: 'Separator' }] })).toThrow('Row components can only contain fields');
+      });
+    });
+    describe('success', () => {
+      it('should do nothing with a correct Page', async () => {
+        expect(() => SmartActionFormLayoutService.validateLayoutElement({ component: 'Page', elements: [{ type: 'Layout', component: 'Separator' }] })).not.toThrow('');
+      });
+
+      it('should do nothing with a correct Separator', async () => {
+        expect(() => SmartActionFormLayoutService.validateLayoutElement({ component: 'Separator' })).not.toThrow('');
+      });
+    });
+  });
+
+  describe('extractFieldsAndLayout', () => {
+    it('should extract out the layout and fields from the hook result', async () => {
+      const { smartActionFormLayoutService } = setup();
+
+      const form = [
```

Comment:
This is the type that the client will need to provide ?

Reply:
Yes. It is the same kind of syntax as v2.
We agreed on it with agent team this morning.

Reply:
It could ease the transition to v2 if the syntax is roughly similar

Reply:
Could you add some example without `type: 'Layout',`

Reply:
Done